src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -991,6 +991,29 @@
         (autosign-certificate-request! subject csr settings)
         (fs/delete (path-to-cert-request csrdir subject))))))
 
+(schema/defn ^:always-validate delete-certificate-request!
+  :- {:outcome (schema/enum :success :not-found :error) :message schema/Str}
+  "Delete pending certificate requests for subject"
+  [{:keys [csrdir]} :- CaSettings
+   subject :- schema/Str]
+  (let [csr-path (path-to-cert-request csrdir subject)]
+    (if (fs/exists? csr-path)
+      (if (fs/delete csr-path)
+        (let [msg (str "Deleted certificate request for " subject)]
+          (log/debug msg)
+          {:outcome :success
+           :message msg})
+        (let [msg (format "Path %s exists but could not be deleted" csr-path)]
+          (log/error msg)
+          {:outcome :error
+           :message msg}))
+      (let [msg (format
+                  "No certificate request for %s at expected path %s"
+                  subject csr-path)]
+        (log/info msg)
```

Comment:
I think the logging in this function is good overall, although I'd log this at `error` or `warn` instead of `info`.